src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,51 +1,89 @@
-import React, { useCallback, useEffect, useState } from "react";
-import classNames from "classnames";
+import React, { useEffect } from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import classNames from "classnames";
 
 import { getLinkFromText } from "utils/getLinkFromText";
+import { isDefined } from "utils/types";
+
+import { useShowHide } from "hooks/useShowHide";
 
-import { useChatSidebarControls } from "hooks/chatSidebar";
+import { Banner } from "types/banner";
+
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner?: Banner;
+  announcementForUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  announcementForUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isShownAnnouncementMessage,
```

Comment:
Current naming is weird. Maybe something like this?

```suggestion
    isShown: isAnnouncementMessageVisible,
```

Reply:
771b70a done